wagtail_newsletter/templates/wagtail_newsletter/panels/newsletter_panel.html

```diff
@@ -45,6 +49,70 @@
             {% endif %}
         {% endblock %}
 
+    {% elif campaign.is_scheduled %}
+
+        {% if report %}
+            <p>
+                <b>Status:</b>
+                {{ campaign.status }}
+            </p>
+
+            <p>
+                <b>Send time:</b>
+                {{ report.send_time }}
+                (in {{ report.send_time|timeuntil }}).
+            </p>
+        {% endif %}
+
+        {% if has_action_permission.unschedule_campaign %}
+            <div class="wn-panel--buttons">
+                <button
+                    type="button"
+                    class="button no button-longrunning"
+                    data-a11y-dialog-show="wn-unschedule-dialog"
+                    data-controller="w-progress"
+                    data-action="wn-submit:unschedule_campaign@window->w-progress#activate"
+                >
+                    {% icon name="spinner" %}
+                    Unschedule
+                </button>
+
+                {% fragment as unschedule_dialog_subtitle %}
+                    The campaign is scheduled to send at
+                    {{ report.send_time }} (in {{ report.send_time|timeuntil }}).
+                    This action will unschedule the sending.
```

Comment:
Ah, good catch! I have not considered the scenario of somebody having unsaved changes and clicking "Unschedule". If the user is quick, they won't get the browser confirmation. Does this look right?

<img width="1049" alt="Screenshot 2024-09-04 at 15 54 14" src="https://github.com/user-attachments/assets/7a4d30e8-d2f0-4c40-a4fc-58e33e4b712d">

Reply:
> If the user is quick, they won't get the browser confirmation. Does this look right?

@mgax Unfortunately if the user is _not_ quick enough and trigger the "unsaved changes" mechanism, clicking "Unschedule" and then _cancelling_ the browser reload prompt would leave the "Unschedule" button stuck in a loading state until the `ProgressController` timeouts.

You might be able to avoid this by preventing the browser's dialog (now that we provide a sufficiently clear message). This can be done by preventing the `w-unsaved:confirm` event, like how `ActionController.forceReload` does it:

https://github.com/wagtail/wagtail/blob/50b99c76438d46b694d09d75b14196521ed1250d/client/src/controllers/ActionController.ts#L129-L141

Note that the ability to prevent the event is only available for 6.2 and up, though. You'll need a different approach for older releases, but I'm not sure it's worth fixing.

Reply:
I've applied the fix for Wagtail 6.2.